lib/rest-api.php

```diff
@@ -30,3 +30,51 @@ function gutenberg_register_edit_site_export_controller_endpoints() {
 }
 
 add_action( 'rest_api_init', 'gutenberg_register_edit_site_export_controller_endpoints' );
+
+if ( ! function_exists( 'gutenberg_register_wp_rest_post_types_controller_fields' ) ) {
+	/**
+	 * Adds `template` and `template_lock` fields to WP_REST_Post_Types_Controller class.
+	 */
+	function gutenberg_register_wp_rest_post_types_controller_fields() {
+		register_rest_field(
+			'type',
+			'template',
+			array(
+				'get_callback' => function ( $item ) {
+					$post_type = get_post_type_object( $item['slug'] );
+					if ( ! empty( $post_type ) && ! empty( $post_type->template ) ) {
+						return $post_type->template;
+					}
+					return null;
+				},
+				'schema'       => array(
+					'type'        => 'array',
+					'description' => __( 'The template associated with the post type.', 'gutenberg' ),
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+			)
+		);
+		register_rest_field(
+			'type',
+			'template_lock',
+			array(
+				'get_callback' => function ( $item ) {
+					$post_type = get_post_type_object( $item['slug'] );
+					if ( ! empty( $post_type ) && ! empty( $post_type->template_lock ) ) {
+						return $post_type->template_lock;
+					}
+					return null;
+				},
+				'schema'       => array(
+					'type'        => 'string',
+					'emum'        => array( 'all', 'insert', 'contentOnly' ),
```

Comment:
```suggestion
					'enum'        => array( 'all', 'insert', 'contentOnly' ),
```

Reply:
Also: it seems `template_lock` can be a string or a boolean and `false` is a valid value, according to [this](https://github.com/WordPress/wordpress-develop/blob/5c7f94a6aa979dc70ff935e9006957fe0f262932/src/wp-includes/class-wp-post-type.php#L302), see also [this](https://github.com/WordPress/wordpress-develop/blob/5c7f94a6aa979dc70ff935e9006957fe0f262932/src/wp-includes/blocks/blocks-json.php#L3228).